lib/charms/opensearch/v0/opensearch_distro.py

```diff
@@ -431,15 +432,34 @@ def current(self) -> Node:
                 temperature=current_node.get("attributes", {}).get("temp"),
             )
         except OpenSearchHttpError:
-            # we try to get the most accurate description of the node
-            conf_on_disk = self.config.load("opensearch.yml")
+            # we try to get the most accurate description of the node from the static config
+            conf = self.config.load("opensearch.yml")
+            try:
+                roles = conf["node.roles"]
+            except KeyError:
+                roles = self._charm.opensearch_peer_cm.deployment_desc().config.roles
```

Comment:
Apologies, I missed this in the previous review
```suggestion
                if deployment_desc.start == StartMode.WITH_PROVIDED_ROLES:
                    roles = deployment_desc.config.roles
                else:
                    roles = ClusterTopology.generated_roles()
```